Lombiq.HelpfulLibraries.LinqToDb/LinqToDbQueryExecutor.cs

```diff
@@ -0,0 +1,76 @@
+using Dapper;
+using LinqToDB;
+using LinqToDB.Data;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+using static LinqToDB.Common.Configuration;
+using Sql = LinqToDB.Common.Configuration.Sql;
+
+namespace Lombiq.HelpfulLibraries.LinqToDb
+{
+    public static class LinqToDbQueryExecutor
+    {
+        static LinqToDbQueryExecutor()
+        {
+            // Generate aliases for final projection.
+            Sql.GenerateFinalAliases = true;
+
+            // We need to disable null comparison for joins. Otherwise it would generate a syntax like this:
+            // JOIN Table2 ON Table1.Key = Table2.Key OR Table1.Key IS NULL AND Table2.Key IS NULL
+            Linq.CompareNullsAsValues = false;
+        }
+
+        /// <summary>
+        /// Use this extension method for running LINQ syntax-based DB queries.
+        /// </summary>
+        /// <typeparam name="TResult">The type of results to return.</typeparam>
+        /// <param name="session">A YesSql session whose connection is used instead of creating a new one.</param>
+        /// <param name="query">The <see cref="IQueryable"/> which will be run as a DB query.</param>
+        /// <returns>An <see cref="IEnumerable{T}"/> with the input type.</returns>
+        public static async Task<IEnumerable<TResult>> LinqQueryAsync<TResult>(

```

Comment:
SQLite-tal ezt kapom bármilyen paraméteres query-nél, pl. a Deleted contentnél: https://localhost:44316/Admin/Lombiq.DataTables/Table/Get?providerName=DeletedContentItemDataTableDataProvider&queryId=Company&admin=596284924

![image](https://user-images.githubusercontent.com/1976647/119049817-4efd9380-b9c1-11eb-9eab-1ba7ec808a29.png)

Ez azért van, mert  ez lesz az SQL

```sql
"--  SQLite.MS SQLite
DECLARE @contentType NVarChar(7) -- String
SET     @contentType = 'Company'
DECLARE @contentType_1 NVarChar(7) -- String
SET     @contentType_1 = 'Company'

SELECT
	[deleted].[Id],
	[deleted].[DisplayText],
	[deleted].[Author],
	[deleted].[Owner] as [Owner_1],
	[deleted].[CreatedUtc],
	[deleted].[PublishedUtc],
	[deleted].[ModifiedUtc],
	[deleted].[ContentType],
	[deleted].[Latest],
	[deleted].[Published],
	[deleted].[ContentItemVersionId],
	[deleted].[ContentItemId],
	[deleted].[DocumentId]
FROM
	[ContentItemIndex] [deleted]
		LEFT JOIN (
			SELECT
				[contentItemIndex_1].[ContentItemId]
			FROM
				[ContentItemIndex] [contentItemIndex_1]
			WHERE
				[contentItemIndex_1].[ContentType] = @contentType AND
				([contentItemIndex_1].[Latest] = 1 OR [contentItemIndex_1].[Published] = 1)
			GROUP BY
				[contentItemIndex_1].[ContentItemId]
		) [notDeleted] ON [notDeleted].[ContentItemId] = [deleted].[ContentItemId]
WHERE
	[deleted].[ContentType] = @contentType_1 AND [notDeleted].[ContentItemId] IS NULL
"
```

Viszont SQLite-ban nincs `DECLARE` support.

Lehet, hogy valójában nem támogatja az SQLite-ot? Lásd https://github.com/linq2db/linq2db/issues/1981 szerint ennek jónak kéne lennie. Viszont hasonló hiba: https://github.com/linq2db/linq2db/issues/1878 De állítólag ennek mennie kellene: https://github.com/linq2db/linq2db/issues/2689 Ezt nem értem.

Reply:
Hmm, asszem sejtem mi a probléma, kipróbálom.

Reply:
Átírtam, nézd meg légyszi, hogy mit gondolsz.

Reply:
Ahham, jónak tűnik. Prod DB-én is teszteltem, így is OK.
Köszi, hogy átírtad!

Reply:
Akartam is kérni a prod tesztet, fasza akkor!